### DISCUSSION TITLE
getServerSideProps returns `undefined` on client side transition after adding locales

### DISCUSSION THREAD

**Discussion by abdofola** [vercel/next.js#46733] [Help]
### Summary

for no apparent reason `getServerSideProps` fails to run on client side transition after adding `locales` to the project, therefore my page component crashes unexpectedly because there is a client fetching depends on the data recieved as `props` from `getServerSideProps`.

I have oppened an issue [here](https://github.com/vercel/next.js/issues/46730), please take a look, any help would be appreciated!

### Example

https://github.com/abdofola/Abdallah-Money-Tracker

**Comment by icyJoseph** [comment]
Have you tried upgrading from `12.2.0` to the latest `12.x.x`? I do recall there was some bugs with GSSP around those versions.

- [12.3.4](https://www.npmjs.com/package/next/v/12.3.4) is what you want to try with.

**Reply by abdofola** [reply]
I tried to upgrade to `12.3.4`, nothing seems to workout.

**Reply by icyJoseph** [reply]
We'll figure something out, no worries.

When you transition on the SPA, into a page with GSSP, do you see the request leaving the browser? It is usually some `/_next/data` URL that ends with `.json`

**Reply by abdofola** [reply]
Ok, the moment I added this block of code to `next.config.js`  this is  **_when_**  `gSSP` started to fail
```
i18n: {
    locales: ["ar", "en"],
    defaultLocale: "ar",
  }
```

> When you transition on the SPA, into a page with GSSP, do you see the request leaving the browser? It is usually some `/_next/data` URL that ends with `.json`

yes, I see this in the `network` tab, and the browser `url` gets updated as well.

**Reply by abdofola** [reply]
I'm following the strategy of sub-path routing as per the [docs](https://nextjs.org/docs/advanced-features/i18n-routing#locale-strategies), rather than domain per language.

This how am handling the locales in my `Header ` component if any relevent

```
import React from "react";
import Link from "next/link";
import { useRouter } from "next/router";
import { Home, Document, Icon } from "@components/icons";
// import { useAuth } from "@lib/helpers/hooks";
import { en, ar } from "@locales";
import styles from "./Header.module.css";

const pages = {
  home: { path: "/", Icon: <Home /> },
  statement: { path: "/transactions", Icon: <Document /> },
};

// component
export default function Header({ user }) {
  const router = useRouter();
  const { locale, pathname, query, asPath } = router;
  const translation = locale === "en" ? en : ar;
  const renderedListItems = Object.entries(pages).map(
    ([label, { path, Icon }]) => (
      <li key={label} className={styles.item}>
        <Link href={path}>
          <a className={styles.pageLink} data-active={isActive(path)}>
            <span>{Icon}</span>
            <span>{translation.nav[label]}</span>
          </a>
        </Link>
      </li>
    )
  );
  function isActive(path: string) {
    return pathname === path;
  }

  // setting `dir` and `lang` of the document in accordance to `locale`
  React.useEffect(() => {
    const html = window.document.querySelector("html")!;
    const dir = locale === "en" ? "ltr" : "rtl";
    // set `html` direction whenever locale changes.
    html.dir = dir;
    html.lang = locale!;
  }, [locale]);

  return (
    <nav id="nav" className={styles.nav}>
      {/*  logo  */}
      <Link href="/" >
        <a className="fixed ltr:top-2 rtl:top-1 z-10 sm:static">
          <Icon href="/sprite.svg#logo" className="w-7 h-7 sm:w-10 sm:h-10" />
        </a>
      </Link>
      {/* pages' links */}
      <ul className={styles.list}>
        {renderedListItems}
        {/* login/logout */}
        <li
          id="navLogin"
          className="fixed flex gap-4 justify-end bg-white border-b top-0 left-0 right-0 p-2 sm:static sm:border-none sm:p-0"
        >
          <select
            name="locale"
            defaultValue={locale}
            onChange={async (e) => {
              router.push({ pathname, query }, asPath, {
                locale: e.target.value,
                shallow: true,
              });
            }}
          >
            <option value="en">en</option>
            <option value="ar">ar</option>
          </select>

          <Link href={user ? "/api/logout" : "/login"}>
            <a className="px-2 py-px rtl:text-sm text-gray-500 bg-gray-50 rounded-sm">
              {user ? translation.nav["logout"] : translation.nav["login"]}
            </a>
          </Link>
        </li>
      </ul>
    </nav>
  );
}
```

**Reply by icyJoseph** [reply]
Tried to run your app, but there was quite a lot of setup needed.

If you could make a small reproduction demo, that'd be awesome.

I wonder if the iron-session has some issues with the locales? Could you check if the wrapper is getting called? I looked through their repo and found nothing. 

Could you paste the `/_next/data`, full url? removing parts you consider sensitive though.

**Reply by abdofola** [reply]
I also suspect that the issue has sth to do with `iron-session`, tried to remove the wrapper and just using normal `gSSP` but  the issue persists, page's component recieves `undefined` for any data injected  in `props`.

_sorry_ my bad that you struggled to run the repo locally, I've written poor `readme` file.

**_screenshot of `transactions` page url_**


![Screenshot 2023-03-06 195829](https://user-images.githubusercontent.com/30251052/223192391-09e68119-1a59-4f32-99d0-b6c8b5f2c477.png)

**Reply by icyJoseph** [reply]
Aha, that 308 is suspicious... This means that your request may never be leaving the browser, something like 301, but not quite IIRC.

To release that cache, it is not enough with clearing application data, you gotta do `cmd + shift + r` or `ctrl + shift + r` with the DevTools open... it is a bit annoying, but that'll flush out the cache.

Why would it be cached? Well you have a redirect branch in your code, for when user is missing, I guess that gets saved when you arrive to the page without credentials? 

I see you have it as `permanent: false`, but if you ever did have it once as `true` the browser saves that and no HMR is going to free that, you gotta clear it manually.
